### PR TITLE
[SYCL][Deps]Uplift Windows GPU RT from 8673 to 8778

### PR DESCRIPTION
**Pull request by yanfeng3721**
This PR is used for Windows GPU RT uplift test.

### CODE REVIEW

**Comment by bader** [comment]
@yanfeng3721, please, update LIT tests to fix the regressions.

**Comment by bader** [comment]
@yanfeng3721, please, update two more tests.

**Comment by bader** [comment]
BTW, re-running Jenkins summary job is not necessary. Your new patch updates only LIT tests, so I expect the same results as for the first version of the patch.
To simplify tracking of pre-commit testing results and code review comments, it's recommended to avoid force-pushing.

**Review by bader — APPROVED** [review]

**Comment by yanfeng3721** [comment]
> BTW, re-running Jenkins summary job is not necessary. Your new patch updates only LIT tests, so I expect the same results as for the first version of the patch.
> To simplify tracking of pre-commit testing results and code review comments, it's recommended to avoid force-pushing.

Ok,

**Comment by yanfeng3721** [comment]
Close by mistake, reopen it.

File: buildbot/dependency.conf [hunk 1]
```diff
@@ -7,8 +7,8 @@ ocl_cpu_rt_ver_win=2020.11.8.0.27
 # https://github.com/intel/compute-runtime/releases/tag/20.35.17767
 ocl_gpu_rt_ver=20.35.17767
 # Same GPU driver supports Level Zero and OpenCL:
-# https://downloadmirror.intel.com/29817/a08/igfx_win10_100.8673.zip
-ocl_gpu_rt_ver_win=27.20.100.8673
+# https://downloadmirror.intel.com/29817/a08/igfx_win10_100.8778.zip
```
Comment (vladimirlaz) [inline buildbot/dependency.conf:10]: I got https://www.intel.com/content/www/us/en/404.html when try to download RT
Please fix the link.

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: buildbot/dependency.conf (hunk 1 above)
Comment (yanfeng3721) [reply buildbot/dependency.conf:10]: Hi Vladimir, it is expected. The Windows GPU RT 8778 is not published yet. The ETA is WW40.2

**Comment by bader** [comment]
@yanfeng3721, there are failures on Windows.
This looks like a CI bug - `sycl-win-x64-pr` job uses old driver instead of new one.
http://ci.llvm.intel.com:8010/#/builders/18/builds/4879/steps/9/logs/stdio

Platform [#3]           : 
    Profile             : FULL_PROFILE
    Version             : OpenCL 2.1 
    Name                : Intel(R) OpenCL HD Graphics
    Vendor              : Intel(R) Corporation
    Devices             : 1
    Device [#1]         : 
        Type            : GPU
        Profile         : FULL_PROFILE
        Version         : OpenCL 2.1 NEO 
        Name            : Intel(R) UHD Graphics 630
        Vendor          : Intel(R) Corporation
        C version       : OpenCL C 2.0 
        Driver version  : 27.20.100.8673

Please, report this issue to CI team.

**Comment by yinyangsx** [comment]
> @yanfeng3721, there are failures on Windows.
> This looks like a CI bug - `sycl-win-x64-pr` job uses old driver instead of new one.
> http://ci.llvm.intel.com:8010/#/builders/18/builds/4879/steps/9/logs/stdio
> 
> Platform [#3] :
> Profile : FULL_PROFILE
> Version : OpenCL 2.1
> Name : Intel(R) OpenCL HD Graphics
> Vendor : Intel(R) Corporation
> Devices : 1
> Device [#1] :
> Type : GPU
> Profile : FULL_PROFILE
> Version : OpenCL 2.1 NEO
> Name : Intel(R) UHD Graphics 630
> Vendor : Intel(R) Corporation
> C version : OpenCL C 2.0
> Driver version : 27.20.100.8673
> 
> Please, report this issue to CI team.

Hi @bader, I have rebuild this job on specific worker which has 8778 driver, http://ci.llvm.intel.com:8010/#/builders/18/builds/4881
Link will update when it finish.

**Comment by bader** [comment]
> > @yanfeng3721, there are failures on Windows.
> > This looks like a CI bug - `sycl-win-x64-pr` job uses old driver instead of new one.
> > http://ci.llvm.intel.com:8010/#/builders/18/builds/4879/steps/9/logs/stdio
> > Platform [#3] :
> > Profile : FULL_PROFILE
> > Version : OpenCL 2.1
> > Name : Intel(R) OpenCL HD Graphics
> > Vendor : Intel(R) Corporation
> > Devices : 1
> > Device [#1] :
> > Type : GPU
> > Profile : FULL_PROFILE
> > Version : OpenCL 2.1 NEO
> > Name : Intel(R) UHD Graphics 630
> > Vendor : Intel(R) Corporation
> > C version : OpenCL C 2.0
> > Driver version : 27.20.100.8673
> > Please, report this issue to CI team.
> 
> Hi @bader, I have rebuild this job on specific worker which has 8778 driver, http://ci.llvm.intel.com:8010/#/builders/18/builds/4881
> Link will update when it finish.

@yinyangsx, I expected that buildbot/dependency.conf configures the driver on buildbot worker. Am I wrong?

**Comment by bader** [comment]
@yanfeng3721, http://ci.llvm.intel.com:8010/#/builders/2/builds/5308/steps/15/logs/FAIL__SYCL__asm_no_operands_cpp failure seems to be unrelated to your change. Please, file a bug report to investigate it separately.

**Comment by yinyangsx** [comment]
> > > @yanfeng3721, there are failures on Windows.
> > > This looks like a CI bug - `sycl-win-x64-pr` job uses old driver instead of new one.
> > > http://ci.llvm.intel.com:8010/#/builders/18/builds/4879/steps/9/logs/stdio
> > > Platform [#3] :
> > > Profile : FULL_PROFILE
> > > Version : OpenCL 2.1
> > > Name : Intel(R) OpenCL HD Graphics
> > > Vendor : Intel(R) Corporation
> > > Devices : 1
> > > Device [#1] :
> > > Type : GPU
> > > Profile : FULL_PROFILE
> > > Version : OpenCL 2.1 NEO
> > > Name : Intel(R) UHD Graphics 630
> > > Vendor : Intel(R) Corporation
> > > C version : OpenCL C 2.0
> > > Driver version : 27.20.100.8673
> > > Please, report this issue to CI team.
> > 
> > 
> > Hi @bader, I have rebuild this job on specific worker which has 8778 driver, http://ci.llvm.intel.com:8010/#/builders/18/builds/4881
> > Link will update when it finish.
> 
> @yinyangsx, I expected that buildbot/dependency.conf configures the driver on buildbot worker. Am I wrong?

Windows GPU RT doesn't like others which are in share folder, it has to be installed into machine. So we can't set windows GPU version by modify dependency.conf. The GPU version in dependency.conf just for showing.

**Comment by bader** [comment]
> Windows GPU RT doesn't like others which are in share folder, it has to be installed into machine. So we can't set windows GPU version by modify dependency.conf. The GPU version in dependency.conf just for showing.

Can BuildBot select properly configured worker based on the information from dependency.conf file?

**Comment by yinyangsx** [comment]
> > Windows GPU RT doesn't like others which are in share folder, it has to be installed into machine. So we can't set windows GPU version by modify dependency.conf. The GPU version in dependency.conf just for showing.
> 
> Can BuildBot select properly configured worker based on the information from dependency.conf file?

No, we have to select manually

**Comment by bader** [comment]
> > > Windows GPU RT doesn't like others which are in share folder, it has to be installed into machine. So we can't set windows GPU version by modify dependency.conf. The GPU version in dependency.conf just for showing.
> > 
> > 
> > Can BuildBot select properly configured worker based on the information from dependency.conf file?
> 
> No, we have to select manually

How difficult it would be to automate the selection of correctly configured worker?
We can't trust CI results w/o checking manually consistency of the environment captured in the logs with the dependency.conf.

**Comment by yinyangsx** [comment]
> We can't trust CI results w/o checking manually consistency of the environment captured in the logs with the dependency.conf.

For this issue, we can add a feature to do version checking in 'check_env' stage.

**Comment by bader** [comment]
> > We can't trust CI results w/o checking manually consistency of the environment captured in the logs with the dependency.conf.
> 
> For Windows GPU version, I think we can add a feature to do version checking in 'check_env' stage.

That is not as good as automatic selection of correctly configured worker, but it sounds like an improvement to me.
👍

**Pull request merged**